### ISSUE TITLE
Reducing task stuck at 'Calling task.run() method'

### ISSUE THREAD

**Issue by newturok** [PrefectHQ/prefect#2270]
## Description
Reducing task stuck at 'Calling task.run() method'
![image](https://user-images.githubusercontent.com/5757827/78467464-cf0d5f00-7715-11ea-8128-4bb352909c5f.png)

## Expected Behavior
Reducing task is runned and finnished

## Reproduction
With `['asdfasd'] * 2` it runs correct, as on left screen  
With `['asdfasd'] * 3` it stucks, as on the right
With `['asdfasd'] * 3`  and `flow.run()` it runs correct
![image](https://user-images.githubusercontent.com/5757827/78467531-738fa100-7716-11ea-8a98-86a111f952dc.png)

```python
from prefect import task, Flow, Task
from prefect.engine.result_handlers import JSONResultHandler


class MapTestTask(Task):
    def run(self, text):
        self.logger.warning(text)
        return text


class ReduceTextTask(Task):
    def run(self, text_array):
        output = ', \n'.join(text_array)
        self.logger.warning(output)
        return output


handler = JSONResultHandler()

with Flow("Example", result_handler=handler) as flow:
    my_task = MapTestTask()
    reduce_task = ReduceTextTask()
    map_result = my_task.map(['asdfasd'] * 3)
    result = reduce_task(map_result)


if __name__ == '__main__':
    flow.serialize()
    flow.register()
```

## Environment
Clean conda enviorment with just `conda install -c conda-forge prefect` runned
```json
{
  "config_overrides": {},
  "env_vars": [],
  "system_information": {
    "platform": "Windows-10-10.0.19592-SP0",
    "prefect_version": "0.10.0",
    "python_version": "3.7.7"
  }
}
```

**Comment by joshmeek** [comment]
Not seeing this issue when running on MacOS and I believe [this thread](https://prefect-community.slack.com/archives/CL09KU1K7/p1585897676366500) in the public slack is related. Both seem to be due to runs stuck in a running state on Windows environments

**Comment by vembloud** [comment]
I'm experiencing the same, however the problem in my case is not limited to map/reduce tasks. Even this simple flow won't finish (only 4 out of 8 tasks finish):
```
from prefect import Flow, task

@task
def simple_task():
    return 1

with Flow("flow") as flow:
    a1 = simple_task()
    a2 = simple_task()
    a3 = simple_task()
    a4 = simple_task()
    a5 = simple_task()
    a6 = simple_task()
    a7 = simple_task()
    a8 = simple_task()

flow.register(project_name='Hello, World!')
flow.run_agent(token='...')
```

Other things noticed:
- problem is present with both Prefect Cloud and Local Server
- problem goes away when `show_flow_logs=True` parameter is used with `flow.run()`

**Comment by vembloud** [comment]
![image](https://user-images.githubusercontent.com/3707443/78572757-256ac280-7828-11ea-816e-75b17bdfdbbe.png)

**Comment by joshmeek** [comment]
This should be fixed as of the recent `0.10.2` release
